plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -407,7 +407,7 @@ public boolean isNull(ResultSet resultSet, int columnIndex)
             public long readLong(ResultSet resultSet, int columnIndex)
                     throws SQLException
             {
-                return resultSet.getObject(columnIndex, LocalDate.class).toEpochDay();
+                return requireNonNull(resultSet.getObject(columnIndex, LocalDate.class)).toEpochDay();
```

Comment:
how does it fix the NPE? It is now throws NPE in different place.

nit: add message

Reply:
It doesn't. It ensure that the thrown exception is the same as in JDK 11 (NPE without error message instead of a NPE with a detailed reason)

Reply:
> instead of a NPE with a detailed reason)

that's sounds like working against JDK's improvement

there shouldn't be a need for a null-check. for a null value this method shouldn't be invoked at all (`isNull` should return true)

Reply:
So `getObject` returns null but `isNull` returns false

Reply:
And right now we rely on the fact that NPE is being thrown. That didn't change

Reply:
fine. please add a message for the exception being raised and a code comment explaining why we're doing this (this is hardly obvious)